components/integrations/abstract_outgoing_webhook.tsx

```diff
@@ -1,112 +1,85 @@
-
-        /**
-         * The server error text after a failed action
-         */
-        serverError: PropTypes.string.isRequired,
-
-        /**
-         * The hook used to set the initial state
-         */
-        initialHook: PropTypes.object,
-
-        /**
-         * The async function to run when the action button is pressed
-         */
-        action: PropTypes.func.isRequired,
-
-        /**
-         * Whether to allow configuration of the default post username.
-         */
-        enablePostUsernameOverride: PropTypes.bool.isRequired,
-
-        /**
-         * Whether to allow configuration of the default post icon.
-         */
-        enablePostIconOverride: PropTypes.bool.isRequired,
-    }
+type State= {
+    displayName: string;
+    description: string;
+    contentType: string;
+    channelId: string;
+    triggerWords: string;
+    triggerWhen: number;
+    callbackUrls: string;
+    username: string;
+    iconURL: string;
+    saving: boolean;
+    clientError: null | JSX.Element | string;
+}
 
-    constructor(props) {
+export default class AbstractOutgoingWebhook extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
         super(props);
-
         this.state = this.getStateFromHook(this.props.initialHook || {});
     }
 
-    getStateFromHook = (hook) => {
+    getStateFromHook = (hook: Props['initialHook']) => {
         let triggerWords = '';
-        if (hook.trigger_words) {
+        if (hook?.trigger_words) {
             let i = 0;
             for (i = 0; i < hook.trigger_words.length; i++) {
                 triggerWords += hook.trigger_words[i] + '\n';
             }
         }
 
         let callbackUrls = '';
-        if (hook.callback_urls) {
+        if (hook?.callback_urls) {
             let i = 0;
             for (i = 0; i < hook.callback_urls.length; i++) {
                 callbackUrls += hook.callback_urls[i] + '\n';
             }
         }
 
         return {
-            displayName: hook.display_name || '',
-            description: hook.description || '',
-            contentType: hook.content_type || 'application/x-www-form-urlencoded',
-            channelId: hook.channel_id || '',
+            displayName: hook?.display_name ?? '',
+            description: hook?.description ?? '',
+            contentType: hook?.content_type ?? 'application/x-www-form-urlencoded',
+            channelId: hook?.channel_id ?? '',
             triggerWords,
-            triggerWhen: hook.trigger_when || 0,
+            triggerWhen: hook?.trigger_when ?? 0,
             callbackUrls,
             saving: false,
             clientError: null,
-            username: hook.username || '',
-            iconURL: hook.icon_url || '',
+            username: hook?.username ?? '',
+            iconURL: hook?.icon_url ?? '',
         };
     }
 
-    handleSubmit = (e) => {
-        e.preventDefault();
+    handleSubmit = (e?: React.MouseEvent) => {
```

Comment:
we dont have to have the parameter as optional.

Reply:
<img width="444" alt="image" src="https://user-images.githubusercontent.com/99865318/201027372-5ea71111-151d-46fe-9061-fcd5d85690ad.png">
Similarly, If we do not include the parameter as optional we get this error